packages/core/src/persist.ts

```diff
@@ -1,20 +1,55 @@
-import { join, dirname } from "node:path";
+import { resolve, join, dirname } from "node:path";
 import { mkdirSync, renameSync, existsSync, copyFileSync, readdirSync } from "node:fs";
 import Logger from "./util/logger";
+import * as darwin from "./darwin";
 
 const logger = new Logger("core/persist");
 
-export default function persist(asarPath: string) {
+export default async function persist(asarPath: string) {
   try {
-    if (process.platform === "win32") {
-      persistWin32(asarPath);
+    persistAsar(asarPath);
+
+    if (process.platform === "darwin") {
+      // the asar is at Discord.app/Contents/Resources/app.asar
+      const inferredBundlePath = resolve(join(dirname(asarPath), "..", ".."));
+      await postPersistSign(inferredBundlePath);
```

Comment:
Likely running too early because it's not inside of the hook for the update event. You might be able to get away with making the event emitter async, but I would just turn everything in the darwin file sync.